source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -743,6 +743,9 @@ FilterDataStatus Filter::onData(ProcessorState& state, Buffer::Instance& data, b
       // StopIterationAndWatermark as well to stop the ActiveStream from returning error when the
       // last chunk added to stream buffer exceeds the buffer limit.
       state.setPaused(true);
+      if (state.bodyMode() == ProcessingMode::BUFFERED) {
```

Comment:
Can we add some comment here why this special treatment. I think it makes sense in most scenarios, that in happy path, we will buffer the body anyway, so no need to wait for header-response. 

But in the rare case that header-response terminates the request already: error, body overrides, we may want to do some clarification here that this branch buffers unnecessarily.

Reply:
For BUFFERED mode, the API definition is "
// Buffer the message body in memory and send the entire body at once.
    // If the body exceeds the configured buffer limit, then the
    // downstream system will receive an error.
"

This is regardless how the ext_proc server response will be.

Reply:
Updated the comments.